instrumentation-api/src/main/java/io/opentelemetry/instrumentation/api/cache/internal/weaklockfree/AbstractWeakConcurrentMap.java

```diff
@@ -161,6 +162,22 @@ public V putIfAbsent(K key, V value) {
     return previous == null ? target.putIfAbsent(new WeakKey<>(key, this), value) : previous;
   }
 
+  public V computeIfAbsent(K key, Function<? super K, ? extends V> mappingFunction) {
+    if (key == null || mappingFunction == null) {
+      throw new NullPointerException();
+    }
+    V previous;
+    L lookupKey = getLookupKey(key);
+    try {
+      previous = target.get(lookupKey);
+    } finally {
+      resetLookupKey(lookupKey);
+    }
+    return previous == null
+        ? target.computeIfAbsent(new WeakKey<>(key, this), ignored -> mappingFunction.apply(key))
```

Comment:
I thought this was impossible due to needing to pass the weak key to mapping function but the solution of using a capturing lambda was so simple!